website/institutions/views.py

```diff
@@ -6,11 +6,27 @@
 from modularodm import Q
 from modularodm.exceptions import NoResultsFound
 
+from website.project.model import Node
+
+
 def view_institution(inst_id, **kwargs):
     try:
         inst = Institution.find_one(Q('_id', 'eq', inst_id) & Q('is_deleted', 'ne', True))
     except NoResultsFound:
         raise HTTPError(http.NOT_FOUND)
+
+    # The total number of affiliated nodes for each institution is calculated and cached
+    # in the Institution.num_nodes (aka Node.institution_num_nodes) field.
+    num_nodes = len(Node.find_by_institutions(inst, query=(
```

Comment:
A couple of potential improvements since I'm somewhat cautious now:

- This should check to see if num_nodes is already set on the model and if it's below the threshold. If it's above the threshold, then there's no need to run this. 
- We should make a model method that does this setting, and that method should do a post-commit celery task. The first part of the celery task should check the threshold as well and not bother running the query if it's above the threshold.

So this way, we can't have 300 requests all at once, and once it gets past that point, further queries will be bypassed and eventually not made at all. Also, this will DRY up the code a bit.